components/infobox/wikis/starcraft2/infobox_series_custom.lua

```diff
@@ -91,165 +97,172 @@ function CustomInjector:parse(id, widgets)
 	else
 		--needed for e.g. External Cups Lists
-		local name = args.name or mw.title.getCurrentTitle().text
-		VarDefine('featured', args.featured or '')
-		VarDefine('headtohead', args.headtohead or '')
-		VarDefine('tournament_tier', args.liquipediatier or '')
-		VarDefine('tournament_tiertype', args.liquipediatiertype or args.tiertype or '')
-		VarDefine('tournament_mode', args.mode or '1v1')
-		VarDefine('tournament_ticker_name', args.tickername or name)
-		VarDefine('tournament_shortname', args.shortname or '')
-		VarDefine('tournament_name', name)
-		VarDefine('tournament_abbreviation', args.abbreviation or args.shortname or '')
-		local game = args.game
+		local name = _args.name or mw.title.getCurrentTitle().text
+		Variables.varDefine('featured', _args.featured or '')
+		Variables.varDefine('headtohead', _args.headtohead or '')
+		Variables.varDefine('tournament_tier', _args.liquipediatier or '')
+		Variables.varDefine('tournament_tiertype', _args.liquipediatiertype or _args.tiertype or '')
+		Variables.varDefine('tournament_mode', _args.mode or '1v1')
+		Variables.varDefine('tournament_ticker_name', _args.tickername or name)
+		Variables.varDefine('tournament_shortname', _args.shortname or '')
+		Variables.varDefine('tournament_name', name)
+		Variables.varDefine('tournament_abbreviation', _args.abbreviation or _args.shortname or '')
+		local game = _args.game
 		if game then
 			game = _GAMES[game] ~= nil and _GAMES[game][1] or game
 		end
-		VarDefine('tournament_game', game or '')
-		VarDefine('tournament_type', args.type or '')
-		CustomSeries._setDateMatchVar(args.date, args.edate, args.sdate)
+		Variables.varDefine('tournament_game', game or '')
+		Variables.varDefine('tournament_type', _args.type or '')
+		CustomSeries._setDateMatchVar(_args.date, _args.edate, _args.sdate)
 	end
 end
 
---- Allows for overriding this functionality
 function Series:addToLpdb(lpdbData)
 	Variables.varDefine('tournament_icon', lpdbData.icon)
 	Variables.varDefine('tournament_icon_dark', lpdbData.icondark)
 	return lpdbData
 end
 
 function CustomSeries._setDateMatchVar(date, edate, sdate)
-	date = string.match(date or '', '%d%d%d%d%-%d%d%-%d%d')
-		or string.match(edate or '', '%d%d%d%d%-%d%d%-%d%d')
-		or string.match(sdate or '', '%d%d%d%d%-%d%d%-%d%d') or ''
-	sdate = string.match(date or '', '%d%d%d%d%-%d%d%-%d%d')
-		or string.match(sdate or '', '%d%d%d%d%-%d%d%-%d%d')
-		or string.match(edate or '', '%d%d%d%d%-%d%d%-%d%d') or ''
-
-	VarDefine('date', date)
-	VarDefine('tournament_enddate', date)
-	VarDefine('tournament_startdate', sdate)
+	local endDate = CustomSeries._validDateOr(date, edate, sdate)
+	local startDate = CustomSeries._validDateOr(date, sdate, edate)
+
+	Variables.varDefine('date', endDate)
+	Variables.varDefine('tournament_enddate', endDate)
+	Variables.varDefine('tournament_startdate', startDate)
+end
+
+function CustomSeries._validDateOr(dateString1, dateString2, dateString3)
```

Comment:
```suggestion
function CustomSeries._validDateOr(...)
```

Reply:
(Would obviously need to update the rest of the function to support it)

Reply:
passing it along as a table instead and then in the function looping via `Table.iter.spairs` (normal ipairs not possible due to some of them being nil)

Reply:
I was thinking more in the lines of
```lua
function CustomSeries._validDateOr(...)
	local input = { ... }
	local regexString = '%d%d%d%d%-%d%d%-%d%d' --(i.e. YYYY-MM-DD)

	for i, select('#', ...) do
		local dateString = string.match(input[i] or '', regexString)
		if dateString then
			return dateString
		end
	end
end
```

Reply:
```lua
function CustomSeries._validDateOr(...)
	local regexString = '%d%d%d%d%-%d%d%-%d%d' --(i.e. YYYY-MM-DD)

	for _, input in ipairs({...}) do
		local dateString = string.match(input, regexString)
		if dateString then
			return dateString
		end
	end
end
```
You could try this variant also, if it works, cleaner imo

Reply:
second one only works with `Table.iter.spairs` instead of `ipairs` (as we have nils inbetween)
first one only works if using `for i = 1, select('#', ...) do` instead of `for i, select('#', ...) do`

and both would still need the fallback `return ''` at the end

comparing second one (with `Table.iter.spairs`) to what is currently used i do not see how that is much cleaner:
![Screenshot 2022-05-12 10 19 06](https://user-images.githubusercontent.com/75081997/168025303-e1821f97-46d0-4249-bd77-f3377ed5cfaf.png)
(for the current one we can kick the `or ''` too due to `Table.iter.spairs` already kicking the nils)

Reply:
Did you test the second one? I believe it should work even with nils in the middle.

Reply:
because we are pushing the return values into wikivars and passing nil toi them will error
so either have the fallback return of `''` in the function or add for each wiki var again

Reply:
Separate a function from its a users of it. A function should return values that make sense for it, so the function can be independent. The fallback should be located in the function uses the value